pallets/automation-time/src/lib.rs

```diff
@@ -711,15 +717,16 @@ pub mod pallet {
 									task_id.clone(),
 								),
 						};
-						if Self::is_completed_task(task.execution_times.to_vec()) {
-							Tasks::<T>::remove(task_id);
-						}
 						task_action_weight
 							.saturating_add(T::DbWeight::get().writes(1 as Weight))
 							.saturating_add(T::DbWeight::get().reads(1 as Weight))
 					},
 				};
 
+				if let Some(task) = Self::get_task(task_id) {
```

Comment:
oh interesting. hmm, curious what's actually taking ownership of the task up there. i guess the action to construct NativeTransfer or Notify, it looks like it takes ownership of those fields.

i wonder if we could implement #[derive(Copy, Clone)] on the task struct (https://doc.rust-lang.org/rust-by-example/trait/clone.html), so that we can clone up there. alternatively, if you clone each of the fields that are going into the run_native_transfer_task or run_notify_task, i wonder if it'll let you keep ownership